TESTS/mbed_hal/spi_com/configs.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_SPI_PINS_H
+#define MBED_SPI_PINS_H
+
+#if defined(TARGET_NUCLEO_F429ZI)
```

Comment:
If every target has to add to this config file, and `pins.h`, is this file going to grow too big and a quite messy? Is there a better way to achieve the same function?

Reply:
We had a discussion with @ithinuel about this. I was thinking to use `app_config.json` config file. But as far as I remember @ithinuel pointed out that json is not used our green-tea tests and solution with header files is already used in QSPI tests. This is why we decided to use this solution.

Reply:
I discussed this with @maciejbocianski and QSPI test doesn't use header files for pins definition, but only for flash config. In QSPI pins have been unified across all targets with QSPI feature enabled. So in this case we could also define SPI pins in target's `PinNames.h` file. To run one board test we require two SPI interfaces, so I suggest to add the following definitions:
`SPI0_MOSI`, `SPI0_MISO`, `SPI0_SCK`, `SPI0_CS`
`SPI1_MOSI`, `SPI1_MISO`, `SPI1_SCK`, `SPI1_CS`

cc @c1728p9 @0xc0170 @donatieng @jamesbeyond

Reply:
Looks better than these pins in the header file, to define SPIx pins

Reply:
This sounds a better idea to me!

Reply:
Test has been updated:
- removed header files with pins definition
- provided pins definition in target's `PinNames.h` file
- adapted pin names in the test
- adapted documentation.